app/models/apple/config.rb

```diff
@@ -55,14 +62,28 @@ def self.find_or_build_private_feed(podcast)
     end
 
     def self.build_apple_config(podcast, key)
-      ac = Apple::Config.new
+      if has_apple_config?(podcast)
+        Rails.logger.error("Found existing apple config for #{podcast.title}!")
+        Rails.logger.error("Do you want to continue? (y/N)")
+        raise "Stopping build_apple_config" if $stdin.gets.chomp.downcase != "y"
+      end
+
+      ac = podcast.default_feed.apple_configs.first || Apple::Config.new
       ac.public_feed = podcast.default_feed
       ac.private_feed = find_or_build_private_feed(podcast)
       ac.key = key
 
       ac
     end
 
+    def self.mark_as_delivered!(apple_publisher)
+      apple_publisher.episodes_to_sync.each do |episode|
+        if episode.podcast_container&.needs_delivery? == false
+          episode.feeder_episode.apple_has_delivery!
+        end
+      end
+    end
```

Comment:
This is follow up from #853 
When looking at existing episodes, we need to mark the episode as having delivery, based on the container, so that we don't churn/reupload for all the episodes in the feed.